### PR TITLE
2453 Enhance ToTensor transform to support data with no dim

### LINKED ISSUE #2453: `ToTensor()` transform adds unexpected dim
**Describe the bug**
A bug reported by Hogler:
When using `ToTensor()` transform on a numpy array with 0 dims, it added 1 dim.
The root cause is `np.ascontiguousarray()` added a dim:
```
>>> a = np.ascontiguousarray(2)
>>> 
>>> print(a)
[2]
```
I will try to fix it ASAP.

Thanks.

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #2453 .

### Description
This PR enhanced the `ToTensor` transform to support data without dimension, also added unit tests.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: tests/test_to_tensor.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        test_data = torch.as_tensor([[1, 2], [3, 4]])
+        result = ToTensor()(test_data)
+        torch.testing.assert_allclose(result, test_data)
+        self.assertTupleEqual(result.shape, (2, 2))
+
+    def test_single_input(self):
+        test_data = 5
+        result = ToTensor()(test_data)
+        torch.testing.assert_allclose(result, test_data)
+        self.assertEqual(result.ndim, 0)
```
Comment (wyli) [inline tests/test_to_tensor.py:43]: there is a typo in`test_single_tensor_input`? also the tests could be compact, for example:
```suggestion
        for test_data in (5, np.asarray(5), torch.tensor(5)):
            result = ToTensor()(test_data)
            torch.testing.assert_allclose(result, test_data)
            self.assertEqual(result.ndim, 0)
```

Comment (Nic-Ma) [reply tests/test_to_tensor.py]: Thanks for your suggestion.
Yes, you are right. I updated the PR based on your comments, could you please help take a look again?

Thanks.

**Review by wyli — APPROVED** [review]
Thanks!
